plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/server/N4JSProjectManager.java

```diff
@@ -57,12 +59,12 @@ public void initialize(ProjectDescription description, IProjectConfig pProjectCo
 
 	@Override
 	public Result doInitialBuild(CancelIndicator cancelIndicator) {
-		List<URI> uris = new LinkedList<>();
+		Set<URI> uris = new HashSet<>();
 		for (ISourceFolder srcFolder : this.projectConfig.getSourceFolders()) {
 			ISourceFolderEx srcFolderEx = (ISourceFolderEx) srcFolder;
 			uris.addAll(srcFolderEx.getAllResources());
 		}
-		return doBuild(uris, Collections.emptyList(), Collections.emptyList(), cancelIndicator);
+		return doBuild(new LinkedList<>(uris), Collections.emptyList(), Collections.emptyList(), cancelIndicator);
```

Comment:
https://stackoverflow.com/questions/322715/when-to-use-linkedlist-over-arraylist-in-java/42529652#42529652